File: hashmaps/ransom_note_count.rb
# @param {String} ransom_note
# @param {String} magazine
# @return {Boolean}
def can_construct(ransom_note, magazine)
    ransom_note.split("").uniq.each do |char|
        return false if ransom_note.count(char) > magazine.count(char)
    end
    return true
end

File: hashmaps/ransom_note_count.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
      ransom_count = {}

      for char in ransomNote:
          ransom_count[char] = ransom_count.get(char, 0) + 1

      for char, needed in ransom_count.items():
          if magazine.count(char) < needed:
              return False

      return True